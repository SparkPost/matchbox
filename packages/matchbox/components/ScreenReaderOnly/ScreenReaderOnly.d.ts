export default ScreenReaderOnly;
declare function ScreenReaderOnly({ children, as, id }: {
    children: any;
    as: any;
    id: any;
}): any;
declare namespace ScreenReaderOnly {
    namespace propTypes {
        const children: any;
        const as: any;
        const id: any;
    }
    namespace defaultProps {
        const as_1: string;
        export { as_1 as as };
    }
    const displayName: string;
}
