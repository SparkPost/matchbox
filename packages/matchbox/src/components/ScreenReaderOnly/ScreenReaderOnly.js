import React from 'react';
import PropTypes from 'prop-types';
import styles from './ScreenReaderOnly.module.scss';

const ScreenReaderOnly = ({ children }) => (
  <span className={styles.ScreenReaderOnly}>
    {children}
  </span>
);

ScreenReaderOnly.propTypes = {
  children: PropTypes.node.isRequired
};

export default ScreenReaderOnly;
