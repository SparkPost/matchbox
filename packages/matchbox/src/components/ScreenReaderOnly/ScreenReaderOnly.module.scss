@import '../../styles/config';

.ScreenReaderOnly {
  @include visually-hidden();
}
