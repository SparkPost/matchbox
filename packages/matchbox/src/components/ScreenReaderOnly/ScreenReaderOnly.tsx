import React from 'react';
import styled from 'styled-components';
import { visuallyHidden } from '../../styles/helpers';

const StyledScreenReaderOnly = styled('span')`
  ${visuallyHidden}
`;

export type ScreenReaderOnlyProps = {
  as?: 'span' | 'div';
  id?: string;
  children: React.ReactNode;
};

const ScreenReaderOnly = ({ children, as = 'span', id }: ScreenReaderOnlyProps): JSX.Element => (
  <StyledScreenReaderOnly id={id} as={as}>
    {children}
  </StyledScreenReaderOnly>
);

ScreenReaderOnly.displayName = 'ScreenReaderOnly';
export default ScreenReaderOnly;
