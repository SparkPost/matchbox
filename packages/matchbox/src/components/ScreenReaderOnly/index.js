export { default as ScreenReaderOnly } from './ScreenReaderOnly';
