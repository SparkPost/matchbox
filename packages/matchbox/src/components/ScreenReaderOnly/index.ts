export { default as ScreenReaderOnly } from './ScreenReaderOnly';
export * from './ScreenReaderOnly';
