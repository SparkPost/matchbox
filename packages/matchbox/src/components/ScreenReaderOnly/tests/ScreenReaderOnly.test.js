import React from 'react';
import ScreenReaderOnly from '../ScreenReaderOnly';
import { shallow } from 'enzyme';

describe('ScreenReaderOnly', () => {
  it('Renders according to the snapshot', () => {
    const wrapper = shallow(<ScreenReaderOnly>I am screen reader only content.</ScreenReaderOnly>);

    expect(wrapper).toMatchSnapshot();
  });
});
