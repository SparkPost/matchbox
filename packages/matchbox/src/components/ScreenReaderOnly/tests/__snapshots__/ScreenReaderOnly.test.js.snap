// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenReaderOnly Renders according to the snapshot 1`] = `
<span
  className="ScreenReaderOnly"
>
  I am screen reader only content.
</span>
`;
